tree/dataframe/test/dataframe_resptr.cxx

```diff
@@ -64,3 +70,130 @@ TEST(RResultPtr, ImplConv)
    EXPECT_TRUE(m != nullptr);
    EXPECT_TRUE(hasRun);
 }
+
+TEST(RResultPtr, IsReady)
+{
+   ROOT::RDataFrame d(1);
+   auto p = d.Define("x", "rdfentry_").Sum("x");
+   EXPECT_FALSE(p.IsReady());
+
+   p.GetValue();
+   EXPECT_TRUE(p.IsReady());
+}
+
+TEST(RResultHandle, Ctor)
+{
+   ROOT::RDataFrame df(3);
+   auto df2 = df.Define("x", "float(rdfentry_)");
+   auto r1 = df2.Histo1D<float>("x");
+   auto r2 = df2.Sum<float>("x");
+   auto r3 = df2.Count();
+
+   // Test constructors
+   RResultHandle h1(r1);
+   RResultHandle h2(r2);
+   RResultHandle h3(r3);
+}
+
+TEST(RResultHandle, StdVector)
+{
+   ROOT::RDataFrame df(3);
+   auto df2 = df.Define("x", "float(rdfentry_)");
+   auto r1 = df2.Histo1D<float>("x");
+   auto r2 = df2.Sum<float>("x");
+   auto r3 = df2.Count();
+
+   // Test push_back and emplace_back
+   std::vector<RResultHandle> v1;
+   v1.emplace_back(r1);
+   v1.push_back(r1);
+
+   // Test initializer list
+   std::vector<RResultHandle> v2 = {r1, r2, r3};
+}
+
+TEST(RResultHandle, TriggerRun)
+{
+   ROOT::RDataFrame df(3);
+   auto df2 = df.Define("x", "float(rdfentry_)");
+   auto r1 = df2.Sum<float>("x");
+   auto r2 = df2.Count();
+
+   // Check that no event loops has run
+   EXPECT_EQ(df.GetNRuns(), 0u);
+
+   // Create handle and check that no event loop is triggered
+   RResultHandle h(r1);
+   EXPECT_EQ(df.GetNRuns(), 0u);
+
+   // Trigger event loop
+   h.GetValue<float>();
+   EXPECT_EQ(df.GetNRuns(), 1u);
+
+   // Trigger event loop again via the RResultPtr and check that loop has not run
+   h.GetPtr<unsigned long>();
+   r1.GetValue();
+   EXPECT_EQ(df.GetNRuns(), 1u);
+
+   // Trigger event loop again via the second RResultPtr and check that loop has not run
+   r2.GetValue();
+   EXPECT_EQ(df.GetNRuns(), 1u);
+}
+
+TEST(RResultHandle, IsReady)
+{
+   ROOT::RDataFrame df(3);
+   auto df2 = df.Define("x", "float(rdfentry_)");
+   auto r1 = df2.Sum<float>("x");
+   auto r2 = df2.Count();
+
+   RResultHandle h1(r1);
+   RResultHandle h2(r2);
+
+   // Event loop has not yet run
+   EXPECT_FALSE(h1.IsReady());
+   EXPECT_FALSE(h2.IsReady());
+
+   // Trigger loop and check that event loop is marked as run in both actions
+   h1.GetValue<float>();
+   EXPECT_TRUE(h1.IsReady());
+   EXPECT_TRUE(h2.IsReady());
+}
+
+TEST(RResultHandle, GetPtrOrValue)
+{
+   ROOT::RDataFrame df(3);
+   auto df2 = df.Define("x", "float(rdfentry_)");
+   auto r1 = df2.Sum<float>("x");
+   auto r2 = df2.Count();
+
+   RResultHandle h1(r1);
+   RResultHandle h2(r2);
+
+   EXPECT_EQ(h1.GetPtr<float>(), r1.GetPtr());
+   EXPECT_EQ(h1.GetValue<float>(), r1.GetValue());
+
+   EXPECT_EQ(h2.GetPtr<long long unsigned int>(), r2.GetPtr());
+   EXPECT_EQ(h2.GetValue<long long unsigned int>(), r2.GetValue());
```

Comment:
should also check that we detect mismatches between requested type and actual type of the result value

Reply:
That's now done by throwing if we use the wrong template argument.